### PR TITLE
refactor(components): el-popconfirm icon

### PR DESCRIPTION
**Pull request by adaex**
Please make sure these boxes are checked before submitting your PR, thank you!

- [ ] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [ ] Make sure you are merging your commits to `dev` branch.
- [ ] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

File: docs/en-US/component/popconfirm.md [hunk 1]
```diff
@@ -34,16 +34,16 @@ popconfirm/trigger-event
...
-| iconColor         | Icon color          | String  | —               | #f90             |
-| hideIcon          | is hide Icon        | Boolean | —               | false            |
+| Attribute         | Description         | Type             | Accepted Values | Default        |
+| ----------------- | ------------------- | ---------------- | --------------- | -------------- |
+| title             | Title               | String           | —               | —              |
+| confirmButtonText | Confirm button text | String           | —               | —              |
+| cancelButtonText  | Cancel button text  | String           | —               | —              |
+| confirmButtonType | Confirm button type | String           | —               | Primary        |
+| cancelButtonType  | Cancel button type  | String           | —               | Text           |
+| icon              | Icon Component      | String/Component | —               | QuestionFilled |
```
Comment (sxzz) [inline docs/en-US/component/popconfirm.md:44]:
```suggestion
| icon              | Icon Component      | String / Component | —               | QuestionFilled |
```

File: packages/components/popconfirm/src/popconfirm.ts [hunk 2]
```diff
@@ -21,8 +21,8 @@ export const popconfirmProps = {
     default: 'text',
   },
   icon: {
-    type: String,
-    default: 'el-icon-question',
+    type: [String, Object] as PropType<string | Component>,
```
Comment (sxzz) [inline packages/components/popconfirm/src/popconfirm.ts:24]: How about using `buildProp`?

Comment (adaex) [reply packages/components/popconfirm/src/popconfirm.ts:24]: We can update it in the new PR

**Comment by JeremyWuuuuu** [comment]
@adaex Seems like the icon is not rendered, see screenshot:
image

**Comment by adaex** [comment]
> @adaex Seems like the icon is not rendered, see screenshot:

Forgot to update the example. Fixed it.

**Review by imswk — APPROVED** [review]

**Comment by imswk** [comment]
LGTM

**Pull request merged**